runtime/js/40_write_file.js

```diff
@@ -1,85 +1,54 @@
 // Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
 "use strict";
 ((window) => {
-  const { stat, statSync, chmod, chmodSync } = window.__bootstrap.fs;
-  const { open, openSync } = window.__bootstrap.files;
-  const { build } = window.__bootstrap.build;
-  const {
-    TypedArrayPrototypeSubarray,
-  } = window.__bootstrap.primordials;
+  const core = window.__bootstrap.core;
+  const { abortSignal } = window.__bootstrap;
+  const { pathFromURL } = window.__bootstrap.util;
 
   function writeFileSync(
     path,
     data,
     options = {},
   ) {
     options.signal?.throwIfAborted();
-    if (options.create !== undefined) {
-      const create = !!options.create;
-      if (!create) {
-        // verify that file exists
-        statSync(path);
-      }
-    }
-
-    const openOptions = options.append
-      ? { write: true, create: true, append: true }
-      : { write: true, create: true, truncate: true };
-    const file = openSync(path, openOptions);
-
-    if (
-      options.mode !== undefined &&
-      options.mode !== null &&
-      build.os !== "windows"
-    ) {
-      chmodSync(path, options.mode);
-    }
-
-    let nwritten = 0;
-    while (nwritten < data.length) {
-      nwritten += file.writeSync(TypedArrayPrototypeSubarray(data, nwritten));
-    }
-
-    file.close();
+    core.opSync("op_write_file_sync", {
+      path: pathFromURL(path),
+      data,
+      mode: options.mode,
+      append: options.append ?? false,
```

Comment:
The JS-side options logic has somewhat changed from previous.

```ts
writeFileSync("foo", "bar", { append: "" });
```
would call `openSync("foo", { write: true, create: true, truncate: true })`, ie. append would take the `false` value.

Now the same call will lead to `core.opSync("op_write_file_sync", { path: "foo", data: "bar", append: "" })` call. I'm not sure what `serde` will do with that but on the JS side at least the logic is definitely changed.

I'm not sure if it's desired but I would've personally expected these lines to be:
```ts
append: options.append === true,
create: options.create === true,
```

Reply:
There's lots of stuff in the API that will fail if people don't provide the correct types. I think if someone is providing unexpected values for these then it erroring is probably more ideal than it working in a maybe unexpected way. For example, previously if someone provided `null` for `create` it was the same thing as providing `false`, which is probably unexpected given that `undefined` doesn't work that way.